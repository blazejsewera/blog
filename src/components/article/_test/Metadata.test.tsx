import { Metadata } from '../Metadata'
import { render } from '@solidjs/testing-library'

describe('<Metadata />', () => {
  it('renders properly', () => {
    const author = 'author'
    const date = '2022-01-06'
    const language = 'en-US'
    const license = 'CC BY-SA 3.0'
    const siteName = 'blog.sewera.dev'
    const title = 'title'

    const { container, unmount } = render(() => (
      <Metadata {...{ author, date, language, license, siteName, title }} />
    ))

    expect(container).toMatchSnapshot()

    unmount()
  })
})
