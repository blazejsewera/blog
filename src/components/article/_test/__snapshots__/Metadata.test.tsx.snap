// Vitest Snapshot v1

exports[`<Metadata /> > renders properly 1`] = `
<div>
  <meta
    content="article"
    name="og:type"
  />
  <meta
    content="title"
    name="dcterms.title"
  />
  <meta
    content="title"
    name="og:title"
  />
  <meta
    content="2022-01-06"
    name="created"
  />
  <meta
    content="2022-01-06"
    name="date"
  />
  <meta
    content="2022-01-06"
    name="dcterms.date"
  />
  <meta
    content="2022-01-06"
    name="article:published_time"
  />
  <meta
    content="author"
    name="author"
  />
  <meta
    content="author"
    name="dcterms.creator"
  />
  <meta
    content="author"
    name="article:author"
  />
  <meta
    content="CC BY-SA 3.0 author"
    name="dcterms.rights"
  />
  <meta
    content="CC BY-SA 3.0"
    name="dcterms.license"
  />
  <meta
    content="en-US"
    name="dcterms.language"
  />
  <meta
    content="blog.sewera.dev"
    name="og:site_name"
  />
</div>
`;
